c2rust-transpile/src/c_ast/conversion.rs

```diff
@@ -472,6 +480,30 @@ impl ConversionContext {
             self.visit_node(untyped_context, node_id, new_id, expected_ty)
         }
 
+        // Function declarations' types look through typedefs, but we want to use the types with
+        // typedefs intact in some cases during translation. To ensure that these types exist in the
+        // `TypedAstContext`, iterate over all function decls, compute their adjusted type using
+        // argument types from arg declarations, and then ensure the existence of both this type and
+        // the type of pointers to it.
+        for (_decl_id, located_kind) in self.typed_context.c_decls.iter() {
+            if let kind @ CDeclKind::Function { .. } = &located_kind.kind {
+                let new_kind = self.typed_context.fn_decl_ty_with_declared_args(kind);
```

Comment:
Just to be clear, are we referring to both parameters and arguments as the same thing here?

Reply:
Yes. Other places are not strict about the formal parameter/actual argument distinction.